src/store/DrawToolsStore.js

```diff
@@ -144,6 +144,10 @@ export default class DrawToolsStore {
         return null;
     };
 
```

Comment:
I think we can set up a reaction for this

Let's get rid of `when(() => !!this.context, this.onContextReady);`
and replace it with

```
reaction(() => this.context, () => { 
    if (!!this.context) {
        this.onContextReady();
    } else {
        this.destructor();
    }
});
```

Reply:
Aaah I see! Thanks for the suggestion, I added this.